### PR TITLE
macOS docs and CI build

### PR DESCRIPTION
**Pull request by Baekalfen**
Add package and documentation for building xnvme on macOS. Additionally,
adding macOS as part of GitHub Actions build scripts on pull-requests.

Signed-off-by: Mads Ynddal <m.ynddal@samsung.com>

### CODE REVIEW

File: .github/workflows/verify.yml [hunk 1]
```diff
@@ -329,6 +329,55 @@ jobs:
...
+    runs-on: ${{ matrix.runner.os }}-${{ matrix.runner.ver }}
+    if: (! contains(github.ref, 'ci')) || contains(github.ref, 'ci-build-mac')
+
+    strategy:
+      matrix:
+        runner:
+          - { os: 'macos', ver: '11' }
+
+    steps:
+    - name: Container-preparation, openSUSE does not have tar and gzip...
```
Comment (safl) [inline .github/workflows/verify.yml:346]: You can remove this step

File: scripts/pkgs/macos-11.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+astyle
```
Comment (safl) [inline scripts/pkgs/macos-11.txt:1]: Surprisingly few deps, once SPDK is not taken into consideration.
What about ``clang``? I guess that comes with the user-land? No need to specify installation of it?
I guess ``astyle`` can be removed at this point?

**Review by safl — CHANGES_REQUESTED** [review]
Looks great!
For the git-short-message, since the commit affects multiple "areas", then prefix it like:
```ci,scripts,docs: add build of macOS``

File: .github/workflows/verify.yml (hunk 1 above)
Comment (Baekalfen) [reply .github/workflows/verify.yml]: Done

File: scripts/pkgs/macos-11.txt (hunk 2 above)
Comment (Baekalfen) [reply scripts/pkgs/macos-11.txt]: A working installation of `brew` assumes Xcode and its command line tools are installed. This is what supplies `clang`, and likely a lot of other smaller dependencies.

I've removed `astyle`.

**Review by safl — CHANGES_REQUESTED** [review]
Syntax error in ``verify.yml`` otherwise: LGTM.

File: .github/workflows/verify.yml [hunk 3]
```diff
@@ -302,6 +302,7 @@ jobs:
       uses: actions/download-artifact@v2
       with:
         name: archive-src
+
```
Comment (safl) [inline .github/workflows/verify.yml:305]: Please don't make unrelated stylistic changes.

The reason why there is no empty line is that those two steps go "hand-in-hand" download + extract the source.
One can argue whether or not an empty line should be here, however, that would be a separate commit changing that.
If you feel strongly about the it, then the best way forward would be the addition of some tool forcing certain formatting for the ``.yaml`` or at least a separate commit that one can add to git-blame-ignore.

File: .github/workflows/verify.yml [hunk 4]
```diff
@@ -570,6 +618,7 @@ jobs:
...
```
Comment (safl) [inline .github/workflows/verify.yml:621]: Please don't make unrelated stylistic changes.

The reason why there is no empty line is that those two steps go "hand-in-hand" download + extract the source.
One can argue whether or not an empty line should be here, however, that would be a separate commit changing that.
If you feel strongly about the it, then the best way forward would be the addition of some tool forcing certain formatting for the ``.yaml`` or at least a separate commit that one can add to git-blame-ignore.

File: .github/workflows/verify.yml [hunk 5]
```diff
@@ -348,6 +395,7 @@ jobs:
...
```
Comment (safl) [inline .github/workflows/verify.yml:398]: Please don't make unrelated stylistic changes.

The reason why there is no empty line is that those two steps go "hand-in-hand" download + extract the source.
One can argue whether or not an empty line should be here, however, that would be a separate commit changing that.
If you feel strongly about the it, then the best way forward would be the addition of some tool forcing certain formatting for the ``.yaml`` or at least a separate commit that one can add to git-blame-ignore.

**Review by safl — CHANGES_REQUESTED** [review]
Added some nit-picking comments :)

**Review by safl — APPROVED** [review]
Thanks @Baekalfen ! This is looking great.

**Pull request merged**